vaccine_feed_ingest/runners/nc/myspot_gov/normalize.py

```diff
@@ -0,0 +1,151 @@
+    loc_id = site["Event Location Id"]
+    return f"nc:myspot_gov:{loc_id}"
+
+
+def _get_name(site: dict) -> str:
+    return site["Provider Location Name"]
+
+
+def _get_address(site: dict):
+    return schema.Address(
+        street1=site["Street Address"],
+        street2=site["Street Address 2"],
+        city=site["City"],
+        state=site["State"],
+        zip=site["Postal Code"],
+    )
+
+
+def _get_location(site: dict):
+    if site["latitude"] == "" or site["longitude"] == "":
+        return None
+    return schema.LatLng(
+        latitude=float(site["latitude"]),
+        longitude=float(site["longitude"]),
+    )
+
+
+def _get_contacts(site: dict):
+    return [
+        schema.Contact(
+            phone=site["Appointment Phone"],
+            website=site["Web Address"],
+        )
+    ]
+
+
+def _normalize_date(dt: str):
+    if dt == "":
+        return None
+    return dt[0:4] + "-" + dt[4:6] + "-" + dt[6:8]
+
+
+def _get_opening_dates(site: dict):
+    if site["Start Date"] == "" and site["End Date"]:
+        return None
+    return [
+        schema.OpenDate(
+            opens=_normalize_date(site["Start Date"]),
+            closes=_normalize_date(site["End Date"]),
+        )
+    ]
+
+
+def _get_inventories(site: dict):
+    ret = []
+    if site["Moderna"] == "Y":
+        ret.append(schema.Vaccine(vaccine="moderna", supply_level="in_stock"))
+    if site["Pfizer"] == "Y":
+        ret.append(schema.Vaccine(vaccine="pfizer_biontech", supply_level="in_stock"))
+    if site["Janssen"] == "Y":
+        ret.append(
+            schema.Vaccine(vaccine="johnson_johnson_janssen", supply_level="in_stock")
+        )
+    if site["Moderna"] == "N":
+        ret.append(schema.Vaccine(vaccine="moderna", supply_level="out_of_stock"))
+    if site["Pfizer"] == "N":
+        ret.append(
+            schema.Vaccine(vaccine="pfizer_biontech", supply_level="out_of_stock")
+        )
+    if site["Janssen"] == "N":
+        ret.append(
+            schema.Vaccine(
+                vaccine="johnson_johnson_janssen", supply_level="out_of_stock"
+            )
+        )
+    return ret
+
+
+def _get_organization(site: dict):
+    if site["Organization Name"] == "":
+        return None
+    if site["Organization Name"] == "Walmart, Inc.":
+        return schema.Organization(name=site["Organization Name"], id="walmart")
+    return schema.Organization(name=site["Organization Name"])
+
+
+def _get_notes(site: dict):
+    ret = []
+    ret.append("cvms_scheduling:" + site["CVMS Scheduling"])
+    ret.append("county:" + site["County"])
+    if site["Event Type"] != "" and site["Event Type"] != "Not Applicable":
+        ret.append("event_type:" + site["Event Type"])
+    return ret
+
+
+def _get_source(site: dict, timestamp: str) -> schema.Source:
+    return schema.Source(
+        data=site,
+        fetched_at=timestamp,
+        fetched_from_uri="https://myspot.nc.gov/api/get-vaccine-locations",
+        id=_get_id(site),
```

Comment:
Within the `Source` block, let's use the `id` defined in the data source directly ([see schema](https://github.com/CAVaccineInventory/vaccine-feed-ingest/wiki/Normalized-Location-Schema#specification)).

The "decorated" id which includes the file info is great for the general id of the normalized record!